assets/js/common/ApiKeySettingsModal/ApiKeySettingsModal.stories.jsx

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-console, react/function-component-definition */
+import React from 'react';
+import { useArgs } from '@storybook/client-api';
+import ApiKeySettingsModal from './ApiKeySettingsModal';
+
+export default {
+  title: 'Components/ApiKeySettingsModal',
+  component: ApiKeySettingsModal,
+  argTypes: {
+    open: {
+      description: 'Whether the modal is open or not',
+      control: {
+        type: 'boolean',
+      },
+    },
+    loading: {
+      description: 'Whether the settings are loading or submitting',
+      control: {
+        type: 'boolean',
+      },
+    },
+    generatedApiKeyExpiration: {
+      description:
+        'The new generated api key expiration expressed in ISO8601 Timestamp',
+    },
+    generatedApiKey: {
```

Comment:
you should include the `onGenerated` here with some action. this way, you don't really need to mock it in the args, as storybook will add the action in its record:
```
onGenerate: {
   action: 'Generate key',
   description: 'New key is generated',
}
```

Reply:
done